src/main/java/org/folio/rest/impl/PermissionUtils.java

```diff
@@ -42,5 +62,49 @@ public static boolean equals(OkapiPermission okapiPerm, Permission perm) {
         && Objects.equals(okapiPerm.getDescription(), perm.getDescription())
         && Objects.equals(okapiPerm.getDisplayName(), perm.getDisplayName())
         && Objects.equals(okapiPerm.getSubPermissions(), perm.getSubPermissions());
-  }  
+  }
+
+  /**
+   * Purge deprecated permissions for a given tenant.
+   * 
+   * @param pgClient {@link PostgresClient}
+   * @param tenantId tenant id
+   * @return {@link Future} with {@link PermissionNameListObject}
+   */
+  public static Future<PermissionNameListObject> purgeDeprecatedPermissions(PostgresClient pgClient, String tenantId) {
+    Promise<PermissionNameListObject> promise = Promise.promise();
+    pgClient.startTx(tx -> {
+      try {
+        pgClient.select(tx, String.format(SELECT_DEPRECATED_PERMS, tenantId), res -> {
+          PermissionNameListObject permNames = new PermissionNameListObject();
```

Comment:
No check for success/failure here. A bad tenant (no init = no tables) will make a failures and the res.result() below will be an unhandled NPE resulting in no HTTP response generated.
```suggestion
          if (res.failed()) {
            pgClient.rollbackTx(tx, done -> promise.fail(res.cause()));
            return;
          }
          PermissionNameListObject permNames = new PermissionNameListObject();
```

Reply:
Added. Thanks for the test case.